src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -338,6 +339,27 @@ public Response add(@Context HttpHeaders httpheaders, InputStream stream) throws
 		}
 	}
 
+	@POST
+	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
+	@Path("/datapoints/index")
+	public Response index(@QueryParam("start_absolute") Long startTime, @QueryParam("end_absolute") Long endTime, @QueryParam("metric") String metric) throws InvalidServerTypeException
+	{
+		checkServerType(ServerType.INGEST, "JSON /datapoints/index", "POST");
+
+		try {
+			datastore.indexTags(new QueryMetric(startTime, endTime, 0, "metric"));
```

Comment:
Ideally this TTL is set to the same as the data fetched from the DB. I don't think it should be a query parameter.

Reply:
You talking about the 3rd parameter, cacheTime?  Yeah don't expose that.

Reply:
Is 0 alright to hardcode? This is just for the query, which I don't think we want to use the cache for.